### PR TITLE
Split up forwardKin (CPP) into transformPoints, relativeQuaternion, relativeRollPitchYaw

### PR DESCRIPTION
**Pull request by tkoolen**
Part of https://github.com/RobotLocomotion/drake/issues/1476

This is to improve the user interface and improve efficiency (can return fixed-size matrices instead of dynamic). In addition, it is needed for TrigPoly support, as there should not be any sqrt calls and such used in rotation conversions at compile time.

### CODE REVIEW

**Comment by rdeits** [comment]
This will conflict with #1575 but should be easy to resolve in which ever one gets merged later.

**Comment by tkoolen** [comment]
Yeah, we'll figure it out.

File: drake/systems/plants/RigidBodyTree.h [hunk 1]
```diff
@@ -286,35 +286,44 @@ class DRAKERBM_EXPORT RigidBodyTree
...
 
   template <typename Scalar, typename DerivedPoints> // not necessarily any relation between the two; a major use case is having an AutoDiff KinematicsCache, but double points matrix
-  Eigen::Matrix<Scalar, Eigen::Dynamic, DerivedPoints::ColsAtCompileTime> forwardKin(
-      const KinematicsCache<Scalar> &cache, const Eigen::MatrixBase<DerivedPoints> &points, int current_body_or_frame_ind, int new_body_or_frame_ind, int rotation_type) const
+  Eigen::Matrix<Scalar, 3, DerivedPoints::ColsAtCompileTime> transformPoints(
+      const KinematicsCache<Scalar> &cache, const Eigen::MatrixBase<DerivedPoints> &points, int from_body_or_frame_ind, int to_body_or_frame_ind) const
   {
-    cache.checkCachedKinematicsSettings(false, false, "forwardKin"); // rely on forwardJac for gradient cache check
+    static_assert(DerivedPoints::RowsAtCompileTime == 3 || DerivedPoints::RowsAtCompileTime == Eigen::Dynamic, "points argument has wrong number of rows");
+    auto T = relativeTransform(cache, to_body_or_frame_ind, from_body_or_frame_ind);
```
Comment (tkoolen) [inline drake/systems/plants/RigidBodyTree.h]: I really don't like that relativeTransform uses this ordering of from_body_or_frame_ind and to_body_or_frame_ind. Should I change it? The bad thing is that changing the order of these int arguments won't result in compile errors in other people's code.

File: drake/util/drakeGeometryUtil.h [hunk 2]
```diff
@@ -619,19 +620,19 @@ void angularvel2rpydotMatrix(const Eigen::MatrixBase<DerivedRPY>& rpy,
   Scalar cp = cos(p);
   Scalar tp = sp / cp;
 
-  phi << cy / cp, sy / cp, 0.0, -sy, cy, 0.0, cy * tp, tp * sy, 1.0;
+  phi << cy / cp, sy / cp, Scalar(0), -sy, cy, Scalar(0), cy * tp, tp * sy, Scalar(1);
```
Comment (tkoolen) [inline drake/util/drakeGeometryUtil.h]: The Scalar() thing was necessary to fix an issue with second order AutoDiffScalar derivatives.

**Comment by tkoolen** [comment]
Getting [C2719](https://msdn.microsoft.com/en-us/library/373ak2y1.aspx) on appveyor. Appears to be this bug: http://eigen.tuxfamily.org/bz/show_bug.cgi?id=1006

File: drake/systems/plants/RigidBodyTree.h (hunk 1 above)
Comment (RussTedrake) [inline drake/systems/plants/RigidBodyTree.h]: What about taking this chance to change to passing pointers?  We want to do that everywhere soonish.


On January 9, 2016 at 12:20:01 PM, Twan Koolen (notifications@github.com) wrote:

In drake/systems/plants/RigidBodyTree.h:

>    {
> -    cache.checkCachedKinematicsSettings(false, false, "forwardKin"); // rely on forwardJac for gradient cache check
> +    static_assert(DerivedPoints::RowsAtCompileTime == 3 || DerivedPoints::RowsAtCompileTime == Eigen::Dynamic, "points argument has wrong number of rows");
> +    auto T = relativeTransform(cache, to_body_or_frame_ind, from_body_or_frame_ind);
I really don't like that relativeTransform uses this ordering of from_body_or_frame_ind and to_body_or_frame_ind. Should I change it? The bad thing is that changing the order of these int arguments won't result in compile errors in other people's code.

Comment (tkoolen) [inline drake/systems/plants/RigidBodyTree.h]: Yeah, good idea to combine those method signature changes. But I would prefer to not make all of those changes in this pull request.

**Pull request merged**